Objects/listobject.c

```diff
@@ -2682,16 +2685,34 @@ list_index_impl(PyListObject *self, PyObject *value, Py_ssize_t start,
         if (stop < 0)
             stop = 0;
     }
-    for (i = start; i < stop && i < Py_SIZE(self); i++) {
-        PyObject *obj = self->ob_item[i];
-        Py_INCREF(obj);
-        int cmp = PyObject_RichCompareBool(obj, value, Py_EQ);
-        Py_DECREF(obj);
-        if (cmp > 0)
-            return PyLong_FromSsize_t(i);
-        else if (cmp < 0)
-            return NULL;
-    }
+    if (!key)
+        for (i = start; i < stop && i < Py_SIZE(self); i++) {
```

Comment:
you can add 
`int cmp` before for if stmt

and modify for-loops like that:
`for (i = start; i < stop && i < Py_SIZE(self) && cmp == 0; i++)`

reference: https://github.com/python/cpython/blob/d36a36511899bd8dc9adf5fc354e8ff7c611c51d/Objects/listobject.c#L452

Reply:
that's pretty pointless, also not the way the old code was.  anyway, I've now modified it.